### PR TITLE
feat: add navigation

### PR DESCRIPTION
**Pull request by limsohee1002**
### Description
Add navigation
Add footer
Add theme toggle

### Type(s) of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Update to an existing feature

### Motivation for PR
#12 #10 #9 

### Applicable screenshots
https://www.loom.com/share/1f4a094c82a445e3b28997e7354ea265?sid=d2677004-be98-47e1-9b7e-113a50b14863

## Release Notes

- **New Features**
	- Introduced a `ThemeSwitch` component for toggling between light and dark themes.
	- Added a `Footer` component with structured links and social media integration.
	- Implemented a `Navbar` component with a responsive design and search functionality.
	- Added a collection of icon components for enhanced visual representation.

- **Improvements**
	- Updated navigation and footer configurations for customization.
	- Enhanced typography with new mobile variants for better responsiveness.
	- Improved layout capabilities with new CSS classes and adjustments.
	- Introduced a new CSS file for style resets and default styles.

- **Chores**
	- Updated dependencies in the project to support new features and improvements.

### CODE REVIEW

File: src/css/types.css [hunk 1]
```diff
@@ -6,49 +6,102 @@
...
     font-style: normal;
     font-weight: 300;
-    line-height: 4rem;
+    line-height: 3.84rem;
     letter-spacing: -0.08rem;
     -webkit-font-smoothing: antialiased;
     -moz-osx-font-smoothing: grayscale;
   }
 
+  .serif-heading-0-mobile {
```
Comment (limsohee1002) [inline src/css/types.css:15]: updated type to match updated design

**Comment by limsohee1002** [comment]
Please ignore font:
It seems like docusaurus is not loading the font asset. It adds italic medium weight font asset only, so it render it as italic and medium font weight right now.

File: src/css/infma-overrides.css [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  box-sizing: unset;
+  background-color: unset;
+  margin: unset;
+  padding-block: unset;
+  padding-inline: unset;
+  border-width: unset;
+  border-style: unset;
+  border-color: unset;
+  border-image: unset;
+}
```
[the remark this answers is not in the document]
Comment (mokaymokay) [reply src/css/infma-overrides.css:102]: @limsohee1002 👀 I didn't know you can use `all: unset`

[the remark this answers is not in the document]
Comment (limsohee1002) [reply src/css/infma-overrides.css:102]: Oh nice! I did not know either! I Will look through the coderabbit comment very soon!

**Comment by limsohee1002** [comment]
@nahbee10 ready for review!

**Review by nahbee10 — CHANGES_REQUESTED** [review]
1. noticed that when it's first loaded, the nav bar is light but the bg is with the dark color
Screenshot 2024-11-13 at 3 00 59 PM

2. there's extra padding on the left side of footer list items
Screenshot 2024-11-13 at 3 00 46 PM

**Comment by nahbee10** [comment]
> 1. noticed that when it's first loaded, the nav bar is light but the bg is with the dark color
> 
> 2. there's extra padding on the left side of footer list items
> 
hmm seems like both above not happening on local preview. not sure why! but this latest preview has the issue: https://uniswap-docs-staging-2bkm7493q-uniswap.vercel.app/

**Review by nahbee10 — APPROVED** [review]

**Pull request merged**